src/widgets/header/api/subaccounts.ts

```diff
@@ -30,8 +45,13 @@ const fetchQuery = async (): Promise<AddressView[]> => {
 };
 
 export const useSubaccounts = () => {
+  // Query account balances from view service
+  const { data: balances } = useBalances();
+
   return useQuery({
     queryKey: ['view-service-accounts'],
-    queryFn: fetchQuery,
+    queryFn: () => {
+      return fetchQuery(balances!);
```

Comment:
can't seem to resolve this non-null assertion linting error. Using a nullish coalescing operator (`balances ?? []`) here will cause the widget to break and only the main account in the UI.

Reply:
How about this?

```ts
  return useQuery({
    queryKey: ['view-service-accounts', balances],
    queryFn: () => {
      if (!balances) {
        return []
      }
      return fetchQuery(balances);
    },
  });
```

We should make `balances` a query key btw, that way if a new balances object comes back, we refetch.

Reply:
making balance a query key resolves this